airqo-platform/AirQo-frontend
# [PLAT-477]containerization airqo website

#### Summary of Changes (What does this PR do?)
- This PR adds docker containerization to the current airqo website
#### Is this change ready to hit production in its current state?

#### Status of maturity (all need to be checked before merging):


#### How should this be manually tested?
To test this PR, make sure you have [docker](https://www.docker.com/) installed. Checkout this branch locally and `cd` to `AirQo-frontend/web`.  Get the `.env` from the PR author. Also add the database dump (e.g. airqo_db.sql) in `db_data` folder.

Then run the following steps:
Step 1: Build docker images 
```
docker-compose build
``` 
The command above will build three docker images i.e. `PHP apache server`, `MySQL DB server` and `Adminer` - a web frontend for managing MySQL DB.

Step 2: Start the containers and create a network to connect them together-
```
docker-compose up
```
Once all the containers are fully configured, go to your browser and point it at `http://localhost:8000`. Admin panel is at `http://localhost:8000/admin`.
if you wish to browse the DB, goto `http://localhost:8001`. The credentials are provided in the `.env` file.

Step 3: Clean up
 You can hit `ctrl+c` or `cmd+c`, then, run `docker-compose down`.

**NB: Ensure that the following ports are not be utilize by other services: `8000`, `8001`, `8806`.**

#### What are the relevant tickets?
- [PLAT-477]

#### Screenshots
![Screenshot 2021-04-08 at 10 07 48](https://user-images.githubusercontent.com/15224992/113983191-52462100-9852-11eb-9f08-5fe1555ac927.png)
![Screenshot 2021-04-08 at 10 07 25](https://user-images.githubusercontent.com/15224992/113983220-596d2f00-9852-11eb-8295-4ddf7ebcf3cf.png)
![Screenshot 2021-04-08 at 10 07 13](https://user-images.githubusercontent.com/15224992/113983239-5bcf8900-9852-11eb-9b69-a7aa7fa30f99.png)


[PLAT-477]: https://airqoteam.atlassian.net/browse/PLAT-477
